google/cloud/storage/internal/bucket_requests_test.cc

```diff
@@ -560,6 +560,136 @@ TEST(BucketRequestsTest, GetIamPolicy) {
   EXPECT_THAT(actual, HasSubstr("project-for-billing"));
 }
 
+TEST(BucketRequestsTest, ParseIamPolicyFromString) {
+  nl::json expected_payload{
+      {"kind", "storage#policy"},
+      {"etag", "XYZ="},
+      {"bindings",
+       {
+           // The order of these elements matters, SetBucketIamPolicyRequest
+           // generates them sorted by role. If we ever change that, we will
+           // need to change this test, and it will be a bit more difficult to
+           // write it.
+           nl::json{
+               {"role", "storage.buckets.list"},
```

Comment:
Nit: The `storage.<resource>.<operation>` format is generally used for individual permissions, and each role has a set of permissions.  Roles tend to look like `storage.admin`, `storage.ObjectViewer`, etc.

Doesn't really matter for the test logic, but I'd rather not confuse somebody who's looking at the test code :P